internal/controller/observability/collector.go

```diff
@@ -72,6 +72,17 @@ func ReconcileCollector(context internalcontext.ForwarderContext, pollInterval,
 	}
 	trustedCABundle := collector.WaitForTrustedCAToBePopulated(context.Client, context.Forwarder.Namespace, resourceNames.CaTrustBundle, pollInterval, timeout)
 
+	credCm, err := collector.ReconcileAWSCredentialsConfigMap(context.Client, context.Reader, context.Forwarder.Namespace, context.Forwarder.Name, context.Forwarder.Spec.Outputs, context.Secrets, context.ConfigMaps, ownerRef)
+	if err != nil {
+		log.V(9).Error(err, "collector.ReconcileAWSProfileConfig")
+		return err
+	}
+
+	// Add generated credentials configmap to contexts to be mounted in pod
+	if credCm != nil {
+		context.ConfigMaps[credCm.Name] = credCm
```

Comment:
Chance of collision wiht a spec'd secret name.   We may need an early validation to ensure there is none

Reply:
If we have precedence and are certain there is limited chance of collision then leave it as-is

Reply:
I think we will have precedence because it will overwrite what is spec'd when the data etc does not match